packages/reactive_ble_platform_interface/lib/src/model/discovered_characteristic.dart

```diff
@@ -38,8 +38,8 @@ class DiscoveredCharacteristic {
       (((17 * 37) + characteristicId.hashCode) * 37 + serviceId.hashCode) * 37;
 
   @override
-  bool operator ==(dynamic other) =>
+  bool operator ==(Object other) =>
       runtimeType == other.runtimeType &&
-      characteristicId == other.characteristicId &&
-      serviceId == other.serviceId;
+      characteristicId == (other as DiscoveredCharacteristic).characteristicId &&
```

Comment:
There is some technical debt here, as this cast could fail at runtime. But I am only here to make a no-op change, and not change behavior.

Reply:
@srawlins I came to the same conclusion here.

What if you did

```
@override
bool operator ==(Object other) {
  return other is DiscoveredCharacteristic &&
    runtimeType == other.runtimeType &&
    characteristicId == other. characteristicId &&
    serviceId == other. serviceId;
}
```

Reply:
Then again, now that I can see the static analysis issues, it looks like I must make the suggested change, to avoid `test_types_in_equals`.

Reply:
Done